### PR TITLE
CNV-6071: UI for VM disk hot-plug

### PR DESCRIPTION
**Pull request by avivtur**
Fixes:
https://issues.redhat.com/browse/CNV-6071

Analysis / Root cause:
Exposing an option to add hotplug disks to a running VM via UI.
Exposing an option to persist hotplug disks to a running VM via UI

Solution Description:
Adding API types and calls to add volume and remove volume, and creating/removing the resources for the hotplug disks.
Adjusting UI components to allow hotplugs.

Screen shots / Gifs for design review:
before:
Screenshot from 2021-07-06 13-44-00
Screenshot from 2021-07-06 13-43-20

after:
stop modal:
cnv6071-modal-stop

restart modal:
cnv6071-modal-restart

add disk modal:
cnv6071-modal-new-2
cnv6071-modal-new-1

disks table:
cnv6071-vmDisks

demo:
cnv6071_create_and_restart_demo

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/selectors/disks/hotplug.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    (pDisk) => pDisk.name,
+  );
+  let hotplugsDiskNames;
+  if (isAutoRemove) {
+    hotplugsDiskNames = hotplugDiskNames.filter((disk) => !persistentDiskNames?.includes(disk));
+  } else {
+    hotplugsDiskNames = hotplugDiskNames.filter((disk) => persistentDiskNames?.includes(disk));
+  }
+
+  return hotplugsDiskNames;
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/selectors/disks/hotplug.ts:44]:
```suggestion
    return hotplugDiskNames.filter((disk) => isAutoRemove ?  !persistentDiskNames?.includes(disk)) : persistentDiskNames?.includes(disk));
```

Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/selectors/disks/hotplug.ts]: added, thanks :)

**Comment by yaacov** [comment]
/hold

@avivtur PTAL, looks like a real fail

File: frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx [hunk 2]
```diff
@@ -403,17 +446,12 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
   return (
     <div className="modal-content">
       <ModalTitle>
-        {isEditing ? t('kubevirt-plugin~Edit') : t('kubevirt-plugin~Add')}{' '}
-        {t('kubevirt-plugin~{{type}}', { type: type.toString() })}
+        <Trans t={t} ns="kubevirt-plugin" values={{ type: type.toString() }}>
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx:449]: Trans is used when you have a text with some component, like icon or link,
here you should use `t`, think of the translator and what he/she see when they get to translate this string

File: frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/HotplugFieldLevelHelp.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import { Stack, StackItem } from '@patternfly/react-core';
+import { Trans, useTranslation } from 'react-i18next';
+import { FieldLevelHelp } from '@console/internal/components/utils';
+
+export const HotplugFieldLevelHelp = () => {
+  const { t } = useTranslation();
+  return (
+    <FieldLevelHelp popoverHasAutoWidth>
+      <Trans t={t} ns="kubevirt-plugin">
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/HotplugFieldLevelHelp.tsx:10]: this is not a good use for `trans`, it's just two strings that need translation, the extra context you give the translator by giving him/her the title+text  in the same translation string does is not enough to justify adding the extra components, that makes translation harder.

File: frontend/packages/kubevirt-plugin/src/components/vms/ActionMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Stack, StackItem } from '@patternfly/react-core';
+import { Trans, useTranslation } from 'react-i18next';
+import { K8sResourceCommon } from '@console/internal/module/k8s/types';
+import { getName, getNamespace } from '@console/shared';
+import { RestartDiskAlert } from '../vm-disks/RestartDiskAlert';
+
+export const ActionMessage: React.FC<{
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vms/ActionMessage.tsx:8]: is this a `ActionMessage` for all actions or a disk specific action message ?

File: frontend/packages/kubevirt-plugin/src/constants/vm/storage/disk-bus.ts [hunk 5]
```diff
@@ -8,21 +8,21 @@ export class DiskBus extends SelectDropdownObjectEnum<string> {
     labelKey: 'kubevirt-plugin~virtio',
     // t('kubevirt-plugin~Optimized for best performance. Supported by most Linux distributions. Windows requires additional drivers to use this model')
     descriptionKey:
-      'kubevirt-plugin~Optimized for best performance. Supported by most Linux distributions. Windows requires additional drivers to use this model',
+      'kubevirt-plugin~Hotplug is enabled only for "scsi" interface. Optimized for best performance. Supported by most Linux distributions. Windows requires additional drivers to use this model',
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/constants/vm/storage/disk-bus.ts:11]: the "not supporting hotplug" message should be at the end of the description, 
you are describing what is "virtio" the most important features of this options is that it's' optimized for performance, the fact that it's not supported by hotplug is scondary and should be at the end of the description

Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/constants/vm/storage/disk-bus.ts]: applied

File: frontend/packages/kubevirt-plugin/src/components/vms/ActionMessage.tsx (hunk 4 above)
Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/components/vms/ActionMessage.tsx:8]: basically it applies only for stopping and restarting VMs at the moment

File: frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/HotplugFieldLevelHelp.tsx (hunk 3 above)
Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/HotplugFieldLevelHelp.tsx]: applied

File: frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx (hunk 2 above)
Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx]: applied

**Comment by yaacov** [comment]
/hold
need to run 'yarn i18n'

**Pull request merged**